apps/server/src/modules/provisioning/strategy/tsp/tsp.strategy.ts

```diff
@@ -15,10 +33,43 @@ export class TspProvisioningStrategy extends ProvisioningStrategy {
 		return SystemProvisioningStrategy.TSP;
 	}
 
-	// eslint-disable-next-line @typescript-eslint/no-unused-vars
-	override getData(input: OauthDataStrategyInputDto): Promise<OauthDataDto> {
-		// TODO EW-1004
-		throw new NotImplementedException();
+	override async getData(input: OauthDataStrategyInputDto): Promise<OauthDataDto> {
+		const decodedAccessToken: JwtPayload | null = jwt.decode(input.accessToken, { json: true });
+
+		if (!decodedAccessToken || !decodedAccessToken.sub) {
```

Comment:
It's not necessary to check sub here. The validate call will check that. It's enough to check the `decodedAccessToken` for null.

Reply:
done